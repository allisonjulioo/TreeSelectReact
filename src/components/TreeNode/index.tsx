import React, { Component } from 'react';
import CheckBox from '../CheckBox';

export default class extends Component {
  render() {
    return (
      <label className="tree-node">
        <CheckBox />
        <span>Nome do node</span>
      </label>
    );
  }
}
