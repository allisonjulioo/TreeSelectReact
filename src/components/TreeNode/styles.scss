@import '../../styles/';

.step-node {
  .step-node {
    padding-left: $spacing * 5;
  }
  &__tree {
    height: $spacing * 10;
    padding: 0 $spacing * 5;
    position: relative;
    cursor: pointer;
    display: flex;
    align-items: center;
    user-select: none;
    &:hover {
      background: $grey;
    }
  }
  &__action {
    position: absolute;
    right: $spacing * 3;
    height: 100%;
    width: $spacing * 10;
  }
  &__children {
    transition: all 1s ease-in-out;
    overflow: hidden;
    &.oppened {
      height: auto;
    }
    &.collapsed {
      height: 0px;
    }
  }
}
