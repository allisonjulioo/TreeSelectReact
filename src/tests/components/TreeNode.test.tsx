import React from "react";
import TreeNode from "../../components/TreeNode";
// import data from "./mocks/tree";
import { shallow } from "enzyme";

const renderComponent = (props = {}) => shallow(<TreeNode />);

describe("<TreeNode/>", () => {
  it("should match default snapshot", () => {
    const wrapper = renderComponent();
    expect(wrapper).toMatchSnapshot();
  });
  it("renders the component", () => {
    const component = shallow(<TreeNode />);
    expect(component).toMatchSnapshot();
  });
});
