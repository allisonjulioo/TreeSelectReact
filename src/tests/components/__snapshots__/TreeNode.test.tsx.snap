// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TreeNode/> renders the component 1`] = `
<label
  className="checkbox-input"
>
  <default_1 />
  <span>
    Nome do node
  </span>
</label>
`;

exports[`<TreeNode/> should match default snapshot 1`] = `
<label
  className="checkbox-input"
>
  <default_1 />
  <span>
    Nome do node
  </span>
</label>
`;
